### ISSUE TITLE
CountOf expects Specification

### ISSUE THREAD

**Issue by peter-gribanov** [Happyr/Doctrine-Specification#145]
[CountOf expects Specification](https://github.com/Happyr/Doctrine-Specification/blob/master/src/Spec.php#L340).

Is not good because impossible use filters as specification for CountOf:

```php
$total = (int)$rep->match(
    Spec::сountOf(
        Spec::eq('user', $user->getId()
    ),
    new AsSingleScalar()
);
```

**Comment by peter-gribanov** [comment]
I can make PR for fix it

**Comment by peter-gribanov** [comment]
Now, for use filters as specification for CountOf need wrap the filters:

```php
$total = (int)$rep->match(
    Spec::сountOf(
        Spec::andX(
            Spec::eq('user', $user->getId()
        )
    ),
    new AsSingleScalar()
);
```

**Comment by peter-gribanov** [comment]
Fixed in #149
